File: my-folder/0125-valid-palindrome/solution.py
class Solution(object):
    def isPalindrome(self, s):
        a=''
        for i in s:
            if i.isdigit() or i.isalpha():
                a=a+i.upper()
        
        if a[::-1] == a:
            return True
        else:
            return False

File: my-folder/0125-valid-palindrome/solution.java
class Solution {
    public boolean isPalindrome(String s) {
        int n=s.length();
        int i = 0;
        int j = n-1;
        while(i<j){
            char first = s.charAt(i);
            char last  = s.charAt(j);

            if (!Character.isLetterOrDigit(first)){
                i++;
                continue;
            }   

            if (!Character.isLetterOrDigit(last)){
                j--;
                continue;
            }

            if(Character.toLowerCase(first)!= Character.toLowerCase(last))    return false;
            i++;
            j--;
        }

        return true;
    }
}
